lib/bundler/audit/cli/formats/junit.rb

```diff
@@ -0,0 +1,118 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with bundler-audit.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+require 'thor'
+
+module Bundler
+  module Audit
+    class CLI < ::Thor
+      module Formats
+        module Junit
+          #
+          # Prints any findings as an XML junit report.
+          #
+          # @param [Report] report
+          #   The results from the {Scanner}.
+          #
+          # @param [IO, File] output
+          #   Optional output stream.
+          #
+          def print_report(report, output=$stdout)
+            original_stdout = $stdout
+            $stdout = output
+
+            print_xml_testsuite(report) do
+              report.each do |result|
+                print_xml_testcase(result)
+              end
+            end
+
+            $stdout = original_stdout
+          end
+
+          private
+
+          def print_xml_testsuite(report)
+            say <<~XML
+              <?xml version="1.0" encoding="UTF-8" ?>
+              <testsuites id="#{ Time.now.to_i }" name="Bundle Audit">
+                <testsuite id="Gemfile" name="Ruby Gemfile" failures="#{ report.count }">
+            XML
+
+            yield
+
+            say <<~XML
+                </testsuite>
+              </testsuites>
+            XML
+          end
+
+          def print_xml_testcase(result)
+            case result
+            when Results::InsecureSource
+              say <<~XML
+                  <testcase id="#{ result.source }" name="Insecure Source URI found: #{result.source}">
+                    <failure message="Insecure Source URI found: #{result.source}" type="Unknown"></failure>
+                  </testcase>
+              XML
+            when Results::UnpatchedGem
+              say <<~XML
+                  <testcase id="#{ result.gem.name }" name="#{ bundle_title(result) }">
+                    <failure message="#{ result.advisory.title }" type="#{ result.advisory.criticality }">
+                      Name: #{ result.gem.name }
+                      Version: #{ result.gem.version }
+                      Advisory: #{ advisory_ref(result.advisory) }
+                      Criticality: #{ advisory_criticality(result.advisory) }
+                      URL: #{ result.advisory.url }
+                      Title: #{ CGI.escapeHTML(result.advisory.title) }
+                      Solution: #{ advisory_solution(result.advisory) }
+                    </failure>
+                  </testcase>
+              XML
+            end
+          end
+
+          def bundle_title(result)
+            "#{advisory_criticality(result.advisory).upcase} #{result.gem.name}(#{result.gem.version}) #{CGI.escapeHTML(result.advisory.title)}"
+          end
+
+          def advisory_solution(advisory)
+            unless advisory.patched_versions.empty?
+              "upgrade to #{advisory.patched_versions.join(', ')}"
```

Comment:
`Advisory#patched_versions` will definitely contain the `>` character. Should be XML escaped.